helm/opa-mutator-app/rules/functions.rego

```diff
@@ -107,7 +107,11 @@ n_shifted_values(array_in, n) = array_out {
   array_a := array.slice(array_in, start, end)
   array_b := array.slice(array_in, 0, start)
   shifted := array.concat(array_a, array_b)
-  array_out := array.slice(shifted, 0, n)
+  # TODO: concating the array 3 times here makes sure that it's long enough for 3
+  # Masters in one AZ. However, this should be changed to something dynamic
+  doubled:= array.concat(shifted, shifted)
+  quadrupled:= array.concat(doubled, doubled)
+  array_out := array.slice(quadrupled, 0, n)
```

Comment:
This works for right now but it's ugly and I hate it.
Actually we need a function that repeats the values in array enough times that count(array) >= n but I couldn't figure it out in Rego right now. Ideas?

Reply:
No ideas how to make this better in rego. The algorithm is similar in the `aws-operator` and I think it is good enough that way. It is not pretty. It is not super smart. It works and solves a problem without creating more problems. So that works for me.

Reply:
Well you are right. :)
As soon as we think of something more elegant we can still replace it.